Sources/Sentry/SentryUIApplication.m

```diff
@@ -86,6 +88,148 @@ - (UIApplication *)sharedApplication
+                // we reached the end of the view controller stack and the presenter is
+                // the top view controller.
+                break;
+            }
+
+            [result replaceObjectAtIndex:index withObject:topVC.presentedViewController];
+
+            continue;
+        }
+
+        // The top view controller is meant for navigation and not content
+        if ([self isContainerViewController:topVC]) {
+            NSArray<UIViewController *> *contentViewController =
+                [self relevantViewControllerFromContainer:topVC];
+            if (contentViewController != nil && contentViewController.count > 0) {
+                [result removeObjectAtIndex:index];
+                [result addObjectsFromArray:contentViewController];
+            } else {
+                break;
+            }
+            continue;
+        }
+
+        UIViewController *relevantChild = nil;
+        for (UIViewController *childVC in topVC.childViewControllers) {
+            // Sometimes a view controller is used as container for a navigation controller
+            // If the navigation is occupying the whole view controller we will consider this the
+            // case.
+            if ([self isContainerViewController:childVC]
+                && CGRectEqualToRect(childVC.view.frame, topVC.view.bounds)) {
+                relevantChild = childVC;
+                break;
+            }
+        }
+
+        if (relevantChild != nil) {
+            [result replaceObjectAtIndex:index withObject:relevantChild];
+            continue;
+        }
+
+        index++;
+    }
+
+    return result;
+}
+
+- (BOOL)isContainerViewController:(UIViewController *)viewController
+{
+    return [viewController isKindOfClass:UINavigationController.class] ||
+        [viewController isKindOfClass:UITabBarController.class] ||
+        [viewController isKindOfClass:UISplitViewController.class] ||
+        [viewController isKindOfClass:UIPageViewController.class];
+}
+
+- (nullable NSArray<UIViewController *> *)relevantViewControllerFromContainer:
+    (UIViewController *)containerVC
+{
+    if ([containerVC isKindOfClass:UINavigationController.class]) {
+        return @[ [(UINavigationController *)containerVC topViewController] ];
+    }
+    if ([containerVC isKindOfClass:UITabBarController.class]) {
+        UITabBarController *tbController = (UITabBarController *)containerVC;
+        NSInteger selectedIndex = tbController.selectedIndex;
+        if (tbController.viewControllers.count > selectedIndex) {
+            return @[ [tbController.viewControllers objectAtIndex:selectedIndex] ];
+        } else {
+            return nil;
+        }
+    }
+    if ([containerVC isKindOfClass:UISplitViewController.class]) {
+        UISplitViewController *splitVC = (UISplitViewController *)containerVC;
+        if (splitVC.viewControllers.count > 0) {
+            return [splitVC viewControllers];
```

Comment:
So are these false positives? I see the UI test for split VC. Maybe we're not generating coverage from sample app UI tests? Otherwise we might want to raise this with codecov.

Reply:
Exactly, I don't think we generate coverage from UI Tests.